### ISSUE TITLE
Add encryption

### ISSUE THREAD

**Issue by nhantrn** [XhmikosR/notepad2-mod#133]
I was looking for a notepad with encryption feature and ran into [NotepadCrypt](http://www.andromeda.com/people/ddyer/notepad/NotepadCrypt.html). It's based on Notepad2 but haven't been updated for awhile so it doesn't have some of the features I like in Notepad2-mod.

I'd like to request adding encryption feature to Notepad2-mod. The source code for NotepadCrypt is available in the link above. You could use its encryption implementation for Notepad2-mod.

**Comment by QWp6t** [comment]
lol just a 130,000-line diff, no prob. :trollface:

**Comment by XhmikosR** [comment]
Yeah, I doubt this will ever happen, due to the amount of changes and due to the fact that I personally don't have the need for this. But feel to work on it and submit a pull request and we'll see.

**Comment by RaPeHoff** [comment]
Attatched: sources of current Notepad2-mod version (4.2.25.964), based on latest Scintilla (363) with AES-Encryption
[NotePad2Crypt_SRC.zip](https://github.com/XhmikosR/notepad2-mod/files/95732/NotePad2Crypt_SRC.zip)



.

**Comment by XhmikosR** [comment]
That is not a pull request.

**Comment by RaPeHoff** [comment]
You are right :-)
Here is the pull-request (https://github.com/XhmikosR/notepad2-mod/pull/142)
[need some time to learn Git ... :-)]

**Comment by nhantrn** [comment]
@RaPeHoff Notepad2-crypmod crashes when enter a wrong passphrase/masterkey.

**Comment by RaPeHoff** [comment]
Hi,

i can not reproduce this issue.  Please provide an example text and
passphrase (which shouldn't be a really secret ;-)).

If you think, the text returned on entering a wrong passphrase is a failure
- this is not an error (nor an crash), it is a feature.
So any brute force cracker gets not a simple "wrong password" message, but
he has to analyze the result,
which reduces the possible "attacks per second" by a magnitude.

Best regards,
RaPeHoff

2016-01-31 6:52 GMT+01:00 nhantrn <notifications@github.com>:

**Comment by nhantrn** [comment]
[Video_2016-01-31_164544.zip](https://github.com/XhmikosR/notepad2-mod/files/111607/Video_2016-01-31_164544.zip)
I compiled it with VS2015 community edittion.

**Comment by RaPeHoff** [comment]
You are correct, there is a bug in it.
I found the problem and corrected it.
Now you will also get a MessageBox on entering a wrong Password,
it does not make sense to fill in the encrypted resp. wrong decrypted data.

Maybe i find some time today, to push the correted sourced to Github today
...

Many thanx and  best regards, RaPeHoff



2016-02-01 1:48 GMT+01:00 nhantrn <notifications@github.com>:

**Comment by RaPeHoff** [comment]
Notepad3 fork (https://github.com/rizonesoft/Notepad3) integrates the crypto module (AES).
